src/SFA.DAS.Commitments.Api/Orchestrators/Mappers/CommitmentMapper.cs

```diff
@@ -54,6 +54,16 @@ public CommitmentListItem MapFrom(CommitmentSummary source, CallerType callerTyp
             };
         }
 
+        public Types.Commitment.CommitmentAgreement Map(Domain.Entities.CommitmentAgreement commitmentAgreement)
```

Comment:
Initially I wondered if this belonged in the CommitmentMapper and not a CommitmentAgreementMapper, since the former does a lot of stuff already and it could be argued they are different things. However, if this was put in a new mapper, that becomes another dependency for the orchestrator which already has a growing number, so that wouldn't feel great either.

Reply:
In ProviderApprenticeshipService I did add an AgreementMapper, but that fit nicely with the new AgreementOrchestrator.

Commitments was slightly different, in that the new mapper was required by the existing ProviderOrchestrator, which already used CommitmentMapper and CommitmentAgreement *is* a subset of Commitment, so it kind of fitted.

I agree that the current mappers and orchestrators are unwieldily and we should split them up at some point.